src/auth/session/storage/__tests__/utils.ts

```
@@ -0,0 +1,71 @@
+import {Socket, SocketConnectOpts} from 'net';
+
+export function wait(ms: number): Promise<void> {
+  return new Promise((resolve) => setTimeout(() => resolve(), ms));
+}
+
+export async function poll(
+  func: () => Promise<boolean>,
+  {interval = 100, timeout = 5000} = {},
+) {
+  const start = Date.now();
+  while (true) {
+    const elapsed = Date.now() - start;
+    if (elapsed > timeout) {
+      throw Error('Timeout');
+    }
+
+    try {
+      const success = await func();
+      if (success) return;
+    } catch {
+      /* lol empty */
```

we probably should throw an exception here?

---

```suggestion
```